milestone-3/frontend/src/components/recipepage/recipepage.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import "./recipepage.css";
+
+interface recipePageProps {
+  name: string;
+  desc: string;
+  image: string;
+  ingred: string[];
+  instruc: string[];
+}
+
+export default function RecipePage({
+  name,
+  desc,
+  image,
+  ingred,
+  instruc,
+}: recipePageProps) {
+  const [ingredients, setIngredients] = useState(ingred);
+  const [customIngred, setCustomIngred] = useState("");
+  const [instructions, setInstructions] = useState(instruc);
+  const [customInstruc, setCustomInstruc] = useState("");
+
+  function addIngredient(ingred: String){
+    const requestOptions = {
+      method: 'PUT',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({newIngredient: ingred})
+    }
+
+    fetch(`http://localhost:3001/recipe/${name}/ingredient`, requestOptions)
+      .then((response) => console.log("added ingredient"));
+  }
+
+  function addInstruction(instruc: String){
+    const requestOptions = {
+      method: 'PUT',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({newInstruction: instruc})
+    }
+
+    fetch(`http://localhost:3001/recipe/${name}/instruction`, requestOptions)
+      .then((response) => console.log("added instruction"));
```

Comment:
always good to add the .catch() statement at the end of then chains to handle errors